enhancements/workload-partitioning/wide-availability-workload-partitioning.md

```diff
@@ -0,0 +1,731 @@
+`InstallConfig`. This allows our other supported installer methods, such as
+assisted installer or ztp, a straightforward way to expose this feature.
+
+In order to set the correct `Infrastructure Status` at install time, we'll
+modify the `InstallConfig` to include enums to correctly set the status.
+
+```go
+type InstallConfig struct {
+	...
+	// CPUPartitioning determines if a cluster should be setup for CPU workload partitioning at install time.
+	// When this field is set the cluster will be flagged for CPU Partitioning allowing users to segregate workloads to
+	// specific CPU Sets. This does not make any decisions on workloads it only configures the nodes for it.
+	//
+	// This feature will alter the infrastructure nodes and prepare them for CPU Partitioning and as such can not be changed after being set.
+	//
+	// +kubebuilder:default="None"
+	// +optional
+	CPUPartitioning CPUPartitioningMode `json:"cpuPartitioningMode,omitempty"`
+  ...
+}
+
+
+// CPUPartitioningMode defines how the nodes should be setup for partitioning the CPU Sets.
+// +kubebuilder:validation:Enum=None;AllNodes
+type CPUPartitioningMode string
+
+const (
+	// CPUPartitioningNone means that no CPU Partitioning is on in this cluster infrastructure
+	CPUPartitioningNone CPUPartitioningMode = "None"
+	// CPUPartitioningAllNodes means that all nodes are configured with CPU Partitioning in this cluster
+	CPUPartitioningAllNodes CPUPartitioningMode = "AllNodes"
+)
+```
+
+### Machine Config Operator
+
+Once we have a global flag, and a way to set it at install time, we'll need to
+apply the needed configurations at install time during bootstrap. We will add
+this ability to MCO to generate and maintain the needed configurations before
+`kubelet` and the `api-server` stands up.
+
+We will add to the `kubelet` controller the ability to watch the
+`Infrastructure` resource and if CPU Partitioning is set to `AllNodes` we will
+generate the bootstrap and the controller will maintain the MCs from that point
+on. Things to note, this feature is explicitly designed to not be turned off, as
+such once set we will not remove the MCs.
+
+We will need to support upgrades for Single Node since this feature already
+exists for them but this implementation differs slightly. To avoid needless
+restarts, we will not alter the current configuration if we detect that we are
+in a single node topology and the nodes are already configured for CPU Partitioning.
+
+### Admission Controller
+
+We want to remove the checks in the admission controller which specifically
+verifies that partitioning is only applied to single node topology
+configuration. The design and configuration for any pod modification will remain
+the same, we simply will allow you to apply partitioning on non single node
+topologies.
+
+We will use the global identifier to correctly modify the pod spec with the
+`requests.cpu` for the new `requests[management.workload.openshift.io/cores]`
```

Comment:
Note that "management" in this resource type is related to the specific value of the workload annotation on the pods. It could be any value, though currently only "management" is used. Mentioned in the third paragraph of [Kubelet Changes](https://github.com/openshift/enhancements/blob/master/enhancements/workload-partitioning/management-workload-partitioning.md#kubelet-changes)

Reply:
Good call out, I'll update the wording to capture that information so it's not lost